cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go

```diff
@@ -825,64 +827,59 @@ func TestControllerNodeGroupForNodeLookup(t *testing.T) {
 			nodeGroupMaxSizeAnnotationKey: "2",
 		},
 		lookupSucceeds: true,
-	}} {
-		test := func(t *testing.T, i int, annotations map[string]string, useMachineDeployment bool) {
-			t.Helper()
+	}}
 
-			machineSet, machineDeployment := makeMachineOwner(i, 1, annotations, useMachineDeployment)
+	test := func(t *testing.T, tc testCase, testObjs *clusterTestConfig, node *apiv1.Node) {
+		t.Helper()
 
-			node, machine := makeLinkedNodeAndMachine(i, machineSet.Namespace, v1.OwnerReference{
-				Name: machineSet.Name,
-				Kind: machineSet.Kind,
-				UID:  machineSet.UID,
-			})
+		controller, stop := testObjs.newMachineController(t)
+		defer stop()
 
-			machineObjects := []runtime.Object{
-				machine,
-				machineSet,
-			}
+		ng, err := controller.nodeGroupForNode(node)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
 
-			if machineDeployment != nil {
-				machineObjects = append(machineObjects, machineDeployment)
-			}
+		if ng == nil && tc.lookupSucceeds {
+			t.Fatalf("expected nil from lookup")
```

Comment:
expected non-nil? Assuming `ng == nil` and look up succeeded, I would expect `ng != nil`.

Reply:
> expected non-nil? Assuming `ng == nil` and look up succeeded, I would expect `ng != nil`.

I read this as: if lookup of nodegroup fails (is nil) and it was expected to succeed (tc.lookupSucceeds) then this is a failure.

Reply:
I redid the error messages as they were inverted;